python/dgl/frame.py

```diff
@@ -430,8 +427,43 @@ def __repr__(self):
 
     def __getstate__(self):
         if self.storage is not None:
-            _ = self.data               # evaluate feature slicing
-        return self.__dict__
+            # flush any deferred operations
+            _ = self.data
+        state = self.__dict__.copy()
+        # data pinning does not get serialized, so we need to remove that from
+        # the state
+        state['_data_nd'] = None
+        state['pinned_by_dgl'] = False
+        return state
+
+    def __setstate__(self, state):
+        index = None
+        device = None
+        if 'storage' in state and state['storage'] is not None:
+            assert 'index' not in state or state['index'] is None
+            assert 'device' not in state or state['device'] is None
+        else:
+            # we may have a column with only index information, and that is
+            # valid
+            index = None if 'index' not in state else state['index']
+            device = None if 'device' not in state else state['device']
+        assert 'deferred_dtype' not in state or state['deferred_dtype'] is None
```

Comment:
So information related to `deferred_dtype` will not be serialized?

Reply:
Correct, since we access the data in `__getstate__` should cause deferred_dtype to be set to none. In general we don't want any deferred actions to be serialized.